### PR TITLE
Add explicit names to metrics/trace SPI factories

### LINKED ISSUE #1287: Reduce magic of selecting SpanExporterFactory and friends
See https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1253#discussion_r494221024 and https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1262#discussion_r496878453 (and I think it was brought up once before that too 😄)

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Resolves #1287 

See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-environment-variables.md

### CODE REVIEW

File: javaagent-spi/src/main/java/io/opentelemetry/javaagent/spi/exporter/MetricServer.java [hunk 1]
```diff
@@ -23,4 +24,11 @@
    * @param config The configuration
    */
   void start(MetricProducer producer, Properties config);
+
+  /**
+   * Returns names of metric servers supported by this factory.
```
Comment (trask) [inline javaagent-spi/src/main/java/io/opentelemetry/javaagent/spi/exporter/MetricServer.java:29]: i think it's a little confusing why we support multiple names without a bit of explanation:

```suggestion
   * Returns names of metric servers supported by this factory.
   *
   * <p>Multiple names are useful for enabling a pair of span and metric exporters using the same
   * name, while still having separate names for enabling them individually.
```

File: javaagent-spi/src/main/java/io/opentelemetry/javaagent/spi/exporter/SpanExporterFactory.java [hunk 2]
```diff
@@ -21,4 +22,11 @@
    * @return An implementation of a {@link SpanExporter}
    */
   SpanExporter fromConfig(Properties config);
+
+  /**
+   * Returns names of span exporters supported by this factory.
```
Comment (trask) [inline javaagent-spi/src/main/java/io/opentelemetry/javaagent/spi/exporter/SpanExporterFactory.java:27]: same as above

```suggestion
   * Returns names of span exporters supported by this factory.
   *
   * <p>Multiple names are useful for enabling a pair of span and metric exporters using the same
   * name, while still having separate names for enabling them individually.
```

**Review by trask — APPROVED** [review]
Thanks @pavolloffay!

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/TracerInstaller.java [hunk 3]
```diff
@@ -122,21 +114,11 @@ private static SpanExporterFactory findSpanExporterFactory(String exporterName)
...
 
     for (SpanExporterFactory spanExporterFactory : serviceLoader) {
-      if (spanExporterFactory
-          .getClass()
-          .getSimpleName()
-          .toLowerCase()
-          .startsWith(exporterName(exporterName).toLowerCase())) {
-        return spanExporterFactory;
-      }
+      if (spanExporterFactory.getNames().contains(exporterName)) return spanExporterFactory;
```
Comment (iNikem) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/TracerInstaller.java:117]:
```suggestion
      if (spanExporterFactory.getNames().contains(exporterName)) {
        return spanExporterFactory;
      }
```

**Review by iNikem — APPROVED** [review]

**Pull request merged**